services/web/server/src/simcore_service_webserver/users_handlers.py

```diff
@@ -109,3 +113,67 @@ async def delete_token(request: web.Request):
         raise web.HTTPNoContent(content_type=MIMETYPE_APPLICATION_JSON)
     except TokenNotFoundError as exc:
         raise web.HTTPNotFound(reason=f"Token for {service_id} not found") from exc
+
+
+# me/notifications -----------------------------------------------------------
+
+async def _get_user_notifications(redis_client: aioredis.Redis, user_id: int):
+    notifs = []
+    notif_wildcard = f'user_id={user_id}:notification_id=*'
+    async for scanned_notification_key in redis_client.scan_iter(match=notif_wildcard):
+        if notif_str := await redis_client.get(scanned_notification_key):
+            notif = json.loads(notif_str)
+            notifs.append(notif)
+    notifs.sort(key=lambda n: n["date"])
+    notifs.reverse()
+    return notifs
+
+
+@login_required
+async def get_user_notifications(request: web.Request):
+    redis_client = get_redis_user_notifications_client(request.app)
+    user_id = request[RQT_USERID_KEY]
+    notifs = await _get_user_notifications(redis_client, user_id)
+    # last 10 items only
+    return web.json_response(data={"data": notifs[-10:]})
+
+
+@login_required
+async def post_user_notification(request: web.Request):
+    redis_client = get_redis_user_notifications_client(request.app)
+    # body includes the new notification
+    notif = await request.json()
+    nid = random.randint(100, 1000)
+    notif["id"] = str(nid)
+    notif["read"] = "False"
+    notif_hash_key = f'user_id={notif["user_id"]}:notification_id={nid}'
+    await redis_client.set(notif_hash_key, value=json.dumps(notif))
+    response = web.json_response(status=web.HTTPNoContent.status_code)
+    assert response.status == 204  # nosec
+    return response
+
+
+routes = web.RouteTableDef()
+
+
+@routes.patch(f"/{API_VTAG}/notifications/{{nid}}", name="update_user_notification")
+@login_required
+async def update_user_notification(request: web.Request):
+    redis_client = get_redis_user_notifications_client(request.app)
+    user_id = request[RQT_USERID_KEY]
+    nid = request.match_info["nid"]
+    notif_hash_key = f'user_id={user_id}:notification_id={nid}'
+    print("notif_hash_key", notif_hash_key)
+    if notif_str := await redis_client.get(notif_hash_key):
+        notif = json.loads(notif_str)
+        # body includes a dict with the changes to make
+        body = await request.json()
+        print("old notif", notif)
```

Comment:
`potentially consider logging instead of printing`
... is something people told me in the past when using `print()`

Reply:
Cleaned up, danke!